readthedocs/profiles/views.py

```diff
@@ -169,3 +176,81 @@ class TokenDeleteView(TokenMixin, DeleteView):
 
     def get_object(self, queryset=None):  # noqa
         return self.request.user.auth_token
+
+
+class UserSecurityLogView(PrivateViewMixin, ListView):
+    model = AuditLog
+    template_name = 'profiles/private/security_log.html'
+    days_limit = settings.RTD_DEFAULT_LOGS_RETENTION_DAYS
+
+    def get(self, request, *args, **kwargs):
+        download_data = request.GET.get('download', False)
+        if download_data:
+            return self._get_csv_data()
+        return super().get(request, *args, **kwargs)
+
+    def _get_csv_data(self):
+        headers = [
+            'Date',
+            'User',
+            'Project',
+            'Organization',
+            'Action',
+            'IP',
+            'Browser',
+        ]
+        data = (
+            self._get_queryset()
+            .values_list(
+                'created',
+                'log_user_username',
+                'log_project_slug',
+                'log_organization_slug',
+                'action',
+                'ip',
+                'browser',
+            )
+        )
+        now = timezone.now()
+        days_ago = now - timedelta(days=self.days_limit)
+        filename = 'readthedocs_user_security_logs_{username}_{start}_{end}.csv'.format(
+            username=self.request.user.username,
+            start=timezone.datetime.strftime(days_ago, '%Y-%m-%d'),
+            end=timezone.datetime.strftime(now, '%Y-%m-%d'),
+        )
+        csv_data = [
+            [timezone.datetime.strftime(date, '%Y-%m-%d %H:%M:%S'), *rest]
+            for date, *rest in data
+        ]
+        csv_data.insert(0, headers)
+        return get_csv_file(filename=filename, csv_data=csv_data)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['days_limit'] = self.days_limit
+        context['filter'] = self.filter
+        return context
+
+    def _get_queryset(self):
+        user = self.request.user
+        days_ago = timezone.now() - timedelta(days=self.days_limit)
+        queryset = AuditLog.objects.filter(
+            user=user,
+            action__in=[AuditLog.AUTHN, AuditLog.AUTHN_FAILURE],
+            created__gte=days_ago,
+        )
+        return queryset
+
+    def get_queryset(self):
+        queryset = self._get_queryset()
+        # Set filter on self, so we can use it in the context.
+        # Without executing it twice.
+        self.filter = UserSecurityLogFilter(
+            self.request.GET,
+            queryset=queryset,
+        )
+        # If an invalid ip was passed, dont filter it.
+        # Otherwise django will raise an error when executing the queryset.
+        if self.filter.is_valid():
+            return self.filter.qs
```

Comment:
Yea, we definitely need to return an form error here if it isn't valid. If we can simplify this by using a Charfield, I'm 👍 on that.